plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -243,6 +253,676 @@ public void testFlushPartitionCache()
         computeActual(format("DROP TABLE %s", fullyQualifiedTestTableName));
     }
 
+    @Test
+    public void testPartitionProjectionOnVarCharColumnWithEnumProjectionAndStorageLocationTemplate()
```

Comment:
I don't see a test showcasing that the partition projection ignores content that was not specified in the partition projection configuration.

From AWS Athena documentation for the limitations of the functionality:
https://docs.aws.amazon.com/athena/latest/ug/partition-projection.html

> Queries for values that are beyond the range bounds defined for partition projection do not return an error.

Reply:
It's a default behaviour of Trino Hive Connector. When we point to non existing FS resource we just ignore it.
Thats why i did not add this test case.

Reply:
`hive.ignore-absent-partitions` defaults to `false` AFAIK.

Reply:
This case is tested in:
- testPartitionProjectionOnVarcharColumnWithDateProjectionInterval
Partition projection range is excluding last row so it's not return even condition should grab it.
```
 assertQuery(
                format("SELECT name FROM %s WHERE short_name2 > '2001-01-22 02'", fullyQualifiedTestTableName),
                "VALUES ('CZECH_1'), ('CZECH_2')");
```
- testPartitionProjectionOnVarcharColumnWithDateProjectionIntervalExpression
Partition projection range is excluding dates before NOW-3DAYS so it's not returning row with date from 4 days ago.
```
assertQuery(
                format("SELECT name FROM %s WHERE short_name2 >= %s AND short_name2 <= %s", fullyQualifiedTestTableName, day4AgoFormatted, day1AgoFormatter),
                "VALUES ('POLAND_2'), ('CZECH_1'), ('CZECH_2')");
```